src/sentry/static/sentry/app/views/organizationEventsV2/tags.jsx

```diff
@@ -1,18 +1,88 @@
 import React from 'react';
+import PropTypes from 'prop-types';
+import styled from 'react-emotion';
+import {isEqual} from 'lodash';
 
 import SentryTypes from 'app/sentryTypes';
 import TagDistributionMeter from 'app/components/tagDistributionMeter';
+import withApi from 'app/utils/withApi';
+import withGlobalSelection from 'app/utils/withGlobalSelection';
+import {fetchTags} from './utils';
 
-export default class Tags extends React.Component {
+class Tags extends React.Component {
   static propTypes = {
+    api: PropTypes.object.isRequired,
+    organization: SentryTypes.Organization.isRequired,
     view: SentryTypes.EventView.isRequired,
+    selection: SentryTypes.GlobalSelection,
   };
 
+  state = {
+    isLoading: true,
+    tags: {},
+  };
+
+  componentDidMount() {
+    this.fetchData();
+  }
+
+  componentDidUpdate(prevProps) {
+    if (
+      this.props.view.id !== prevProps.view.id ||
+      !isEqual(this.props.selection, prevProps.selection)
+    ) {
+      this.fetchData();
+    }
+  }
+
+  fetchData() {
+    const {api, organization, view} = this.props;
+    fetchTags(api, organization.slug, view.tags)
+      .then(tags => {
+        this.setState({tags, isLoading: false});
```

Comment:
I see more usage of `loading` over `isLoading` 🚲

Reply:
`isLoading` seems to be used in more recent code, I'm going to leave this as is